File: sampleApp/ios/sampleAppTests/BridgeTest.m
//
//  BridgeTest.m
//  sampleAppTests
//
//  Created by Abigail McPhillips on 22/05/2019.
//

#import <XCTest/XCTest.h>
#import <CavyNativeReporter/CavyNativeReporter.h>

@interface BridgeTest : XCTestCase

@end

// Example test that uses the iOS Cavy Native Reporter.
// Waits on Cavy to finish running tests and passes if error count is 0.
@implementation BridgeTest

- (void)testBridge {
  // Make a new expectation.
  XCTestExpectation *expectation = [[XCTestExpectation alloc] initWithDescription: @"Cavy tests passed"];

  [CavyNativeReporter onFinishWithBlock: ^void(NSDictionary* report) {
    // Pull the error count from the report object.
    long errorCount = [report[@"errorCount"] integerValue];
    // Fail if there are errors.
    if (errorCount > 0) {
      XCTFail(@"Cavy tests had one or more errors");
    }
    // Fulfill the expectation.
    [expectation fulfill];
  }];

  // Wait for expectation to fulfill.
  [self waitForExpectations:@[expectation] timeout:100];
}

@end

File: sampleApp/android/app/src/androidTest/java/com/sampleapp/BridgeTest.java
package com.sampleapp.bridgetest;

import androidx.test.rule.ActivityTestRule;

import org.junit.Rule;
import org.junit.Test;
import static org.junit.Assert.*;

import com.cavynativereporter.RNCavyNativeReporterModule;
import com.sampleapp.MainActivity;

// Example test that uses the Android Cavy Native Reporter.
// Waits on Cavy to finish running tests and passes if error count is 0.
public class BridgeTest {

  @Rule
  public ActivityTestRule<MainActivity> activityRule = new ActivityTestRule(MainActivity.class);

  @Test
  public void testBridge() throws Exception {
    // Wait 60 seconds to receive a test report from Cavy.
    RNCavyNativeReporterModule.waitForReport(60);
    // Pull the error count from the report object.
    double errorCount = RNCavyNativeReporterModule.cavyReport.getDouble("errorCount");
    // Note: Third argument is the `delta` allowed between the actual and
    // expected double value.
    assertEquals(errorCount, 0.0, 0.0);
  }
}
